File: whatmakesaman/3주차/PG_86491_최소_직사각형.py
def solution(sizes):
    answer = 0
    
    for size in sizes:
        
        #한 변에 큰 것들 모으고, 다른 변에 작은 것들 모으고
        if size[0]>size[1]:
            size[0], size[1]=size[1],size[0]   
    #최댓값 갱신        
    max_w=max(size[0] for size in sizes)
    max_h=max(size[1] for size in sizes)
    answer=max_w*max_h
    
    return answer

File: kitttae/3주차/PG_86491_최소직사각형.java
import java.util.*;

class Solution {
    public int solution(int[][] sizes) {
        int w = 0;
        int h = 0;
        int maxW = 0;
        int maxH = 0;
        
        for (int[] cur : sizes) {
            w = Math.max(cur[0], cur[1]);
            h = Math.min(cur[0], cur[1]);

            maxW = Math.max(maxW, w);
            maxH = Math.max(maxH, h);
        }
        
        return  maxW * maxH;
    }
}
